### PR TITLE
Adding WithBlacklist option to Exporter to omit health checks, etc.

### PR DESCRIPTION
**Pull request by jeremysf**
Spans who's name match a pattern in the blacklist will not be exported to AWS X-Ray. The purpose of this is to support omitting traces like health checks to reduce clutter in AWS X-Ray and to save costs.

There is precedent for this kind of feature in several of the AWS X-Ray APIs (i.e. rules and whitelist in Node.js API, blacklist in Python API).

### CODE REVIEW

**Comment by jeremysf** [comment]
I signed it!

**Comment by jeremysf** [comment]
Hello! First time Google open source contributor here. Not sure how to get the CLA check to pass. I signed the agreement for my company. In order to do that, my email (jeremy@projector.ai) had to be in the group (google-contributors@projector.ai) and my github account (jeremysf) has that email associated with it (jeremy@projector.ai). Thanks for any help and advice!

**Comment by jeremysf** [comment]
I signed it!

**Comment by jeremysf** [comment]
Thought maybe it was because the commits themselves have the author "Jeremy Gordon <jeremy.gordon@gmail.com>" in them. So I added that email address to the Google Group: google-contributors@projector.ai, but didn't seem to matter. Still failing the CLA check,

**Comment by rakyll** [comment]
Can you squash and recommit the commit?

```
git commit --amend --author="John Doe <john@doe.org>"
```

File: xray.go [hunk 1]
```diff
@@ -128,6 +129,13 @@ func WithBufferSize(bufferSize int) Option {
 	})
 }
 
+// WithBlacklist - spans with name's matching patterns in the blacklist will not be exported
```
Comment (rakyll) [inline xray.go]: // WithBlacklist filters out the spans those names match one of the given regexps.

File: xray.go [hunk 2]
```diff
@@ -279,12 +289,23 @@ func NewExporter(opts ...Option) (*Exporter, error) {
 
 		buffer: buffer,
 		origin: string(c.origin),
+
```
Comment (rakyll) [inline xray.go]: Remove the arbitrary line.

**Comment by jeremysf** [comment]
Cool, made the two changes mentioned in the feedback. Thanks for the guidance on the squash and amend author, that worked great! CLA check passing now. Yay!

**Comment by jeremysf** [comment]
Whoops, had missed addressing one of the comments. Let me know if there's anything else needed.

**Pull request merged**